### PR TITLE
M8P RTK solution support

### PR DESCRIPTION
**Pull request by rrr6399**
Here is an initial cut at exposing the RTK solution from the M8P GPS.

### CODE REVIEW

**Comment by Kiwa21** [comment]
Would  you explain a bit more what you are achieving with this PR ? I though M8P RTK was already supported?

**Comment by rrr6399** [comment]
@Kiwa21 The UBlox driver does not return the GPS_RTK message:
http://mavlink.org/messages/common#GPS_RTK
This is important to understand the relative position of the rover to the base, especially if the base is moving.

**Comment by rrr6399** [comment]
BTW, not all of the parameters, like iar_num_hypotheses, are populated right now since they are missing from the u-blox messaging unfortunately.

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
 //	Origin code by Michael Smith, Jordi Munoz and Jose Julio, DIYDrones.com
 //  Substantially rewritten for new GPS driver structure by Andrew Tridgell
 //
+#include <bitset>
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp]: You can't use this on all the platforms we use and support. (Notably does not work on ChibiOS/ARM GCC 6.3/ARM GCC 4.9) You will need to directly do masking/bit manipulation.

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 2]
```diff
@@ -851,6 +866,59 @@ AP_GPS_UBLOX::_parse_gps(void)
         state.horizontal_accuracy = 0;
 #endif
         break;
+
+    case MSG_NAV_RELPOSNED: {
+        Debug("MSG_NAV_RELPOSNED RTK status Received");
+
+        state.rtk_age_ms    = 0xFFFFFFFF;
+
+    	state.rtk_baseline_x_mm = _buffer.rtk.rel_pos_n_cm*10+_buffer.rtk.rel_pos_hp_n_mm;
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp]: This should be checking the `relPosValid` flag before copying any data. If we timeout with no real data for more then X seconds we should clear out the struct and unset the supports_mavlink_gps_rtk_message stuff.

Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp]: Also note that the HP mm components are actually scaled by 10 (IE reported in a range of -99 to 99) you need to divide them by 10 before adding)

File: libraries/AP_GPS/AP_GPS_UBLOX.cpp [hunk 3]
```diff
@@ -851,6 +866,59 @@ AP_GPS_UBLOX::_parse_gps(void)
...
+    	state.rtk_baseline_y_mm = _buffer.rtk.rel_pos_e_cm*10+_buffer.rtk.rel_pos_hp_e_mm;
+    	state.rtk_baseline_z_mm = _buffer.rtk.rel_pos_d_cm*10+_buffer.rtk.rel_pos_hp_d_mm;
+
+    	int32_t acc_n_mm = _buffer.rtk.acc_n_mm;
+    	int32_t acc_e_mm = _buffer.rtk.acc_e_mm;
+    	int32_t acc_d_mm = _buffer.rtk.acc_d_mm;
+
+    	// rms of error terms
+    	state.rtk_accuracy = safe_sqrt(acc_n_mm*acc_n_mm + acc_e_mm*acc_e_mm + acc_d_mm*acc_d_mm);
```
Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS_UBLOX.cpp]: @amilcarlucas accuracy remains unit less on the MAVLink message which is continuing to give headaches here. @rrr6399's solution of providing a 3D cm accuracy (again these mm values are scaled by a factor of 10 before being reported) is reasonable, but there is no way for a GCS/someone reviewing the code to really know what it is...

**Comment by rrr6399** [comment]
@WickedShell  Thanks for the great feedback! I'll get those changes done as soon as I have time.

**Comment by rrr6399** [comment]
Looks like some platforms don't support the safe_sqrt function.

**Comment by amilcarlucas** [comment]
This nees some squashing

**Pull request closed without merging**